File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/entities/AbstractEvent.java
package com.byagowi.persiancalendar.entities;

import androidx.annotation.NonNull;

import com.byagowi.persiancalendar.calendar.AbstractDate;

abstract public class AbstractEvent<T extends AbstractDate> {
    T date;
    String title;
    boolean holiday;

    public String getTitle() {
        return title;
    }

    public boolean isHoliday() {
        return holiday;
    }

    @NonNull
    public String toString() {
        return title;
    }

    public T getDate() {
        return date;
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/entities/AbstractEvent.kt
package com.byagowi.persiancalendar.entities

import io.github.persiancalendar.calendar.AbstractDate

abstract class AbstractEvent<T : AbstractDate> {
    lateinit var date: T
    open lateinit var title: String
    var isHoliday: Boolean = false

    override fun toString(): String = title
}
